gateway/src/main/java/discord4j/gateway/IdentifyOptions.java

```diff
@@ -46,11 +48,13 @@
      *
      * @param shardInfo shard index and count the client using this policy will identify with
      * @param initialStatus initial presence status the bot will identify with
+     * @param intents the raw intent value to subscribe to
      * @param guildSubscriptions whether to enable presence and typing events while identifying
      */
-    public IdentifyOptions(ShardInfo shardInfo, @Nullable StatusUpdate initialStatus, boolean guildSubscriptions) {
+    public IdentifyOptions(ShardInfo shardInfo, @Nullable StatusUpdate initialStatus, long intents, boolean guildSubscriptions) {
```

Comment:
Guild subscriptions would become deprecated when intents go live. I wonder if we should keep both as separate constructors. @Discord4J/members thoughts?

Depends on the timing of intents and if they would be compatible with v6 gateway.

Reply:
As far as I know, Intents will be fully compatible with v6, so we would be able to omit the guild subscriptions then.

Reply:
I imagine guild subscriptions will eventually be dropped.

Reply:
> While not deprecated, Guild Subscriptions have been superceded by Gateway Intents. You may continue to use guild subscriptions, but we recommend migrating to intents for even better results.

[source](https://github.com/discordapp/discord-api-docs/blob/feature/gateway-intents/docs/topics/Gateway.md#guild-subscriptions)